test/test_torch.py

```diff
@@ -10382,6 +10337,108 @@ def run_test(tensor_dims, some):
+                self.assertEqual(res[0], t.gather(dim, res[1]))
+                if TEST_NUMPY and size % 2 == 1:
+                    # We can only test agains numpy for odd reductions because numpy
+                    # returns the mean of the two medians and torch returns the lower
+                    self.assertEqual(res[0].cpu().numpy(), np.median(t_numpy, dim, keepdims=True))
+
+    @dtypes(torch.float, torch.double)
+    @dtypesIfCUDA(torch.half, torch.float, torch.double)
+    def test_median_nan_values(self, device, dtype):
+        # Generate random 0-3D sizes
+        sizes = [torch.randint(1, 32, (i,)) for i in range(4) for _ in range(2)]
+        for size in sizes:
+            # Create random input tensor with nan values
+            t = torch.rand(size.tolist(), device=device, dtype=dtype)
+            t.masked_fill_(t < 0.1, float('nan'))
+            t_numpy = t.cpu().numpy() if TEST_NUMPY else None
+            for op in [torch.median, torch.nanmedian]:
+                res = op(t)
+                num_nan = t.isnan().sum()
+                if op == torch.median and num_nan > 0:
+                    k = t.numel() - 1
+                else :
+                    k = int((t.numel() - num_nan - 1) / 2)
+                self.assertEqual(res, t.view(-1).sort()[0][k])
+                if TEST_NUMPY and (t.numel() - num_nan) % 2 == 1:
+                    # We can only test agains numpy for odd reductions because numpy
+                    # returns the mean of the two medians and torch returns the lower
+                    numpy_op = np.median if op == torch.median else np.nanmedian
+                    self.assertEqual(res.item(), numpy_op(t.cpu().numpy()))
+                for dim in range(t.ndim):
+                    res = op(t, dim, True)
+                    size = t.size(dim) if t.ndim > 0 else 1
+                    num_nan = t.isnan().sum(dim, True)
+                    if op == torch.median:
+                        k = torch.where(num_nan > 0, size - 1, int((size - 1) / 2))
+                    else:
+                        k = ((size - num_nan - 1) / 2).type(torch.long)
+                    self.assertEqual(res[0], (t.sort(dim)[0]).gather(dim, k))
+                    self.assertEqual(res[0], t.gather(dim, res[1]))
+                    if TEST_NUMPY:
+                        # We can only test agains numpy for odd reductions because numpy
+                        # returns the mean of the two medians and torch returns the lower
+                        numpy_op = np.median if op == torch.median else np.nanmedian
+                        mask = (size - num_nan) % 2 == 1
+                        res = res[0].masked_select(mask).cpu()
+                        ref = numpy_op(t_numpy, dim, keepdims=True)[mask.cpu().numpy()]
+                        self.assertEqual(res, torch.from_numpy(ref))
+
+    def test_median_corner_cases(self, device):
```

Comment:
What about another test for the handling of tensors with no elements?

Reply:
That is tested in test_reduction_empty(). BTW, all the ops in there with no identity need to be updated to follow the new convention for empty tensors.

Reply:
For now the test can assert they fail. File a follow-up issue describing the need for an update and put the link in a comment on the test.

Reply:
It's already testing that it fails. I added a comment linking to https://github.com/pytorch/pytorch/issues/34907 which is the most recently active issue on this.